src/cli/serve/serve.js

```diff
@@ -51,20 +49,31 @@ module.exports = function (program) {
   )
   .option('--plugins <path>', 'an alias for --plugin-dir', pluginDirCollector)
   .option('--dev', 'Run the server with development mode defaults')
-  .option('--no-watch', 'Prevent watching, use with --dev to prevent server restarts')
-  .action(function (opts) {
-    if (opts.dev && opts.watch && !isWorker) {
-      // stop processing the action and handoff to watch cluster manager
-      return require('../watch/watch')(opts);
+  .option('--no-watch', 'Prevents automatic restarts of the server in --dev mode')
```

Comment:
I don't think this is being used anywhere now - can we just drop no-watch from the optimizer?